spec/Section 6 -- Execution.md

```diff
@@ -815,38 +820,81 @@ MergeSelectionSets(fields):
 ### Handling Execution Errors
 
 An _execution error_ is an error raised from a particular field during value
-resolution or coercion. While these errors should be reported in the response,
-they are "handled" by producing a partial response.
+resolution or coercion. These errors should be reported in the response and are
+"handled" according to the selected _error behavior_.
 
-Note: This is distinct from a _request error_ which results in a response with
-no data.
+Note: An _execution error_ is distinct from a _request error_ which results in a
+response with no data.
 
-If an execution error is raised while resolving a field (either directly or
-nested inside any lists), it is handled as though the position at which the
-error occurred resulted in {null}, and the error must be added to the {"errors"}
-list in the response.
+If the result of resolving a field is {null}, and that field is of a `Non-Null`
+type, then an execution error is raised by the field.
 
-If the result of resolving a _response position_ is {null} (either due to the
-result of {ResolveFieldValue()} or because an execution error was raised), and
-that position is of a `Non-Null` type, then an execution error is raised at that
-position. The error must be added to the {"errors"} list in the response.
+If a `List` type wraps a `Non-Null` type, and one of the elements of that list
+resolves to {null}, then an execution error is raised by the list item.
+
+:: The _error behavior_ is the way in which the request wishes for errors to be
+handled. Valid values are {"PROPAGATE"}, {"NO_PROPAGATE"} and {"ABORT"}; their
+respective behaviors are detailed below.
+
+Implementations are free to choose the default value to use if _error behavior_
+is not specified. It is recommended this is {"NO_PROPAGATE"} for newly created
+schemas, but {"PROPAGATE"} should be used for existing schemas since that was
+the implicit behavior in previous versions of this specification.
```

Comment:
Should we allow implementation to not support all the options?

Reply:
No need, any implementation of GraphQL already has the choice to reject any request for any reason. I also don't see any particular value in this, the implementation of `ABORT` for example is literally 3 lines of code:

https://github.com/graphql/graphql-js/blob/1cff42169aaa149c22408c03e099443b887bd229/src/execution/execute.ts#L628-L630
https://github.com/graphql/graphql-js/blob/1cff42169aaa149c22408c03e099443b887bd229/src/error/ErrorBehavior.ts#L7

(A couple more if you include the type safety.)

Reply:
Sounds good 👍 TIL and it makes perfect sense, thanks for the explanation.